src/algebra/homology/local_cohomology.lean

```diff
@@ -0,0 +1,41 @@
+/-
+Copyright (c) 2023 Emily Witt. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Emily Witt, Scott Morrison
+-/
+import ring_theory.ideal.basic
+import algebra.category.Module.colimits
+import algebra.category.Module.projective
+import category_theory.abelian.ext
+
+/-!
+# Local cohomology.
+
+We just give one definition, deferring for later the other equivalent definitions,
+and basic properties.
+-/
+
+open opposite
+open category_theory
+open category_theory.limits
+
+noncomputable theory
+
+variables (R : Type) [comm_ring R] (I : ideal R) (M : Module.{0} R)
```

Comment:
It is possible to make this slightly more universe polymorphic by replacing `Type` by  `Type u` and `Module.{0}` by `Module.{u}` everywhere, provided that some changes are done also in `algebra/category/Module/projective.lean`: the ending of this file could be replaced by something like
```lean
/-- The category of modules has enough projectives, since every module is a quotient of a free
    module. -/
lemma Module_enough_projectives' : enough_projectives (Module.{max u v} R) :=
{ presentation :=
  λ M,
  ⟨{ P := Module.of R (M →₀ R),
    projective := projective_of_free finsupp.basis_single_one,
    f := finsupp.basis_single_one.constr ℕ id,
    epi := (epi_iff_range_eq_top _).mpr
      (range_eq_top.2 (λ m, ⟨finsupp.single m (1 : R), by simp [basis.constr]⟩)) }⟩, }

instance Module_enough_projectives : enough_projectives (Module.{max u v} R) :=
  Module_enough_projectives'

instance Module_enough_projectives'' : enough_projectives (Module.{u} R) :=
  Module_enough_projectives'.{u}
```

Reply:
You can merge the first two declarations, right?

Reply:
Yes, the following would be better:
```lean
/-- The category of modules has enough projectives, since every module is a quotient of a free
    module. -/
instance Module_enough_projectives : enough_projectives (Module.{max u v} R) :=
{ presentation :=
  λ M,
  ⟨{ P := Module.of R (M →₀ R),
    projective := projective_of_free finsupp.basis_single_one,
    f := finsupp.basis_single_one.constr ℕ id,
    epi := (epi_iff_range_eq_top _).mpr
      (range_eq_top.2 (λ m, ⟨finsupp.single m (1 : R), by simp [basis.constr]⟩)) }⟩, }

instance Module_enough_projectives' : enough_projectives (Module.{u} R) :=
  Module.Module_enough_projectives.{u}
```

Reply:
Thanks, updated!